### PR TITLE
feat(cmf): http middleware support action.cmf.http structure

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

if you are doing an http request the TYPE must be in 'GET', 'POST', ...
you can't provide custom type.
We want to make our type more explicit (like 'MyComponent.didMount' or 'MyComponent.click.addButton', ...) so the logs are more readable

**What is the chosen solution to this problem?**

add support for the structure in the middleware

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

File: packages/cmf/src/middlewares/http/middleware.js [hunk 1]
```diff
@@ -122,40 +122,47 @@ export const httpMiddleware = ({ dispatch }) => next => (action) => {
...
+		const errorObject = {
+			name: error.name,
+			message: error.description || error.message,
+			number: error.number,
+			stack: error.stack,
+		};
+		const clone = get(error, 'stack.response.clone');
+		if (!clone) {
+			dispatch({
+				type: 'HTTP_REDUCE_ERROR',
```
Comment (jlame) [inline packages/cmf/src/middlewares/http/middleware.js]: what is this used for ?

File: packages/cmf/src/middlewares/http/middleware.js [hunk 2]
```diff
@@ -122,40 +122,47 @@ export const httpMiddleware = ({ dispatch }) => next => (action) => {
...
+				error: errorObject,
+				action: httpAction,
+			});
+		} else {
+			// clone the response object else the next call to text or json
+			// triggers an exception Already use
```
Comment (sdwvit) [inline packages/cmf/src/middlewares/http/middleware.js]: sorry I'm having really hard times understanding what do you mean by this comment, could you rephrase?

**Comment by jsomsanith** [comment]
2 Approvals but we'll wait for @jmfrancois to answer the questions and merge if everything is ok

File: packages/cmf/src/middlewares/http/middleware.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ export const DEFAULT_HTTP_HEADERS = {
 };
 
 export function isHTTPRequest(action) {
-	return action.type in HTTP_METHODS;
+	return action.type in HTTP_METHODS || !!get(action, 'cmf.http');
```
Comment (talend-glorieux) [inline packages/cmf/src/middlewares/http/middleware.js]: I'd rather see a regular if conditional than a weird double negative statement.

File: packages/cmf/src/middlewares/http/middleware.js (hunk 1 above)
Comment (jmfrancois) [inline packages/cmf/src/middlewares/http/middleware.js]: at the moment when we have a JS error in a middleware it's hidden by can't read clone attribute of undefined.
So let's dispatch that error so we may display the error 'to the end user or not'

May I just throw the error ?

File: packages/cmf/src/middlewares/http/middleware.js (hunk 3 above)
Comment (jmfrancois) [inline packages/cmf/src/middlewares/http/middleware.js]: not sure to understand. will catch you today

File: packages/cmf/src/middlewares/http/middleware.js (hunk 2 above)
Comment (jmfrancois) [inline packages/cmf/src/middlewares/http/middleware.js]: it was here before I must admit me too
